src/models/chan.js

```diff
@@ -183,21 +179,86 @@ Chan.prototype.getFilteredClone = function(lastActiveChannel, lastMessage) {
 	}, {});
 };
 
-function writeUserLog(client, msg) {
-	if (!msg.isLoggable()) {
-		return false;
+Chan.prototype.writeUserLog = function(client, msg) {
+	this.messages.push(msg);
+
+	// Does this user have logs disabled
+	if (!client.config.log) {
+		return;
+	}
+
+	// Are logs disabled server-wide
+	if (Helper.config.messageStorage.length === 0) {
+		return;
+	}
+
+	// Is this particular message or channel loggable
+	if (!msg.isLoggable() || !this.isLoggable()) {
+		return;
 	}
 
+	// Find the parent network where this channel is in
 	const target = client.find(this.id);
 
 	if (!target) {
-		return false;
+		return;
 	}
 
-	userLog.write(
-		client.name,
-		target.network.host, // TODO: Fix #1392, multiple connections to same server results in duplicate logs
-		this.type === Chan.Type.LOBBY ? target.network.host : this.name,
-		msg
-	);
-}
+	// TODO: Something more pluggable
+	if (Helper.config.messageStorage.includes("sqlite")) {
+		client.messageStorage.index(target.network.uuid, this.name, msg);
+	}
+
+	if (Helper.config.messageStorage.includes("text")) {
+		userLog.write(
+			client.name,
+			target.network.host, // TODO: Fix #1392, multiple connections to same server results in duplicate logs
```

Comment:
Your PR description says the first commit fixes #1392, is it still TODO?

Reply:
It fixes it in the case of sqlite, I didn't change it for text logging because that would change the logging folder (which is breaking, and out of the scope of this PR).

Reply:
Ok, then we either need to reword the PR description, or remember to re-open the issue after this is merged, as it will auto-close #1392.